builtin/providers/rundeck/resource_private_key.go

```diff
@@ -0,0 +1,132 @@
+package rundeck
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/apparentlymart/go-rundeck-api/rundeck"
+)
+
+func resourceRundeckPrivateKey() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateOrUpdatePrivateKey,
+		Update: CreateOrUpdatePrivateKey,
+		Delete: DeletePrivateKey,
+		Exists: PrivateKeyExists,
+		Read:   ReadPrivateKey,
+
+		Schema: map[string]*schema.Schema{
+			"path": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to the key within the key store",
+				ForceNew:    true,
+			},
+
+			"key_material_file": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to a file containing the private key material to store",
+				// Although the user provides the file, we actually
+				// persist a hash of the file's contents in the state so
+				// we can detect if it changes without disclosing the
+				// private key itself into the state.
+				StateFunc: PrivateKeyMaterialStateFunc,
+			},
```

Comment:
For this one I was trying to avoid the private key ending up in the terraform state, but I guess I could achieve that by hashing it.

I designed it this way mainly for parity with the way ``connection`` accepts a private key, since I expect most folks will use the same private key for both rundeck and provisioning. That is, by the way, why I made the *public* key resource take the key material inline, to match up with the way e.g. ``aws_keypair`` accepts a key.

I can certainly change this to be an inline thing that gets hashed in the config if you think that's the right thing, and consistency with the ``connection`` blocks is not a priority.